transaction.go

```diff
@@ -0,0 +1,71 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+)
+
+// A Transaction is used to push data from one matrix server to another matrix
+// server.
+type Transaction struct {
+	// The ID of the transaction.
+	TransactionID TransactionID `json:"transaction_id"`
+	// The server that sent the transaction.
+	Origin ServerName `json:"origin"`
+	// The server that should receive the transaction.
+	Destination ServerName `json:"destination"`
+	// The millisecond posix timestamp on the origin server when the transction
+	// was created.
+	OriginServerTS Timestamp `json:"origin_server_ts"`
+	// The IDs of the most recent transactions sent by the server.
+	// Mutliple transactions can be sent in parallel so there may be
+	// more than one previous transaction.
+	PreviousIDs []TransactionID `json:"previous_ids"`
+	// The room events pushed by this transaction.
+	PDUs []Event `json:"pdus"`
+	// The ephemeral events pushed by this transaction.
+	EDUs []EDU `json:"edus"`
+}
+
+// A TransactionID identifies a transaction sent by a matrix server to another
+// matrix server. The ID must be unique amoungst the transactions sent from the
+// origin server to the destination, but doesn't have to be globally unique.
```

Comment:
Yes it needs to go in a URL path segment. Yes you can put smiley poo in a URL path segment. But you probably shouldn't.